mypy/plugins/default.py

```diff
@@ -417,3 +419,17 @@ def int_pow_callback(ctx: MethodContext) -> Type:
         else:
             return ctx.api.named_generic_type('builtins.float', [])
     return ctx.default_return_type
+
+
+def int_neg_callback(ctx: MethodContext) -> Type:
+    """Infer a more precise return type for int.__neg__.
+
+    This is mainly used to infer the return type as LiteralType
+    if the original underlying object is a LiteralType object
+    """
+    if isinstance(ctx.type, Instance) and isinstance(ctx.type.last_known_value, LiteralType):
+        value = ctx.type.last_known_value.value
+        fallback = ctx.type.last_known_value.fallback
+        if isinstance(value, int):
+            return LiteralType(-value, fallback)
+    return ctx.default_return_type
```

Comment:
For consistency, you should also check `if isinstance(ctx.type, LiteralType)` as Michael said in the issue.